lib/semian/mysql2.rb

```diff
@@ -53,11 +55,19 @@ def semian_identifier
     end
 
     def query(*args)
-      acquire_semian_resource(adapter: :mysql, scope: :query) { raw_query(*args) }
+      if query_whitelisted?(*args)
+        raw_query(*args)
+      else
+        acquire_semian_resource(adapter: :mysql, scope: :query) { raw_query(*args) }
+      end
     end
 
     private
 
+    def query_whitelisted?(sql, *)
+      QUERY_WHITELIST.any? { |pattern| sql =~ pattern }
```

Comment:
I really don't think that this is a hotpath worth optimizing for. If we have some data that proves me wrong I'm happy to change this, but for now a regexp seems more reasonable. Especially since it's not trivial to correct match this without a regexp. Even the one I have here I could poke holes into... `RELEASE/**/SAVEPOINT` is totally valid SQL but slips through the cracks.. And so is `RELEASE        SAVEPOINT`...